homeassistant/components/essent/__init__.py

```diff
@@ -0,0 +1,40 @@
+"""The Essent integration."""
+
+from __future__ import annotations
+
+from homeassistant.const import Platform
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers import config_validation as cv
+from homeassistant.helpers.typing import ConfigType
+
+from .const import DOMAIN
+from .coordinator import EssentConfigEntry, EssentDataUpdateCoordinator
+
+PLATFORMS: list[Platform] = [Platform.SENSOR]
+CONFIG_SCHEMA = cv.config_entry_only_config_schema(DOMAIN)
+
+
+async def async_setup(hass: HomeAssistant, config: ConfigType) -> bool:
+    """Set up the Essent integration."""
+    return True
+
+
+async def async_setup_entry(hass: HomeAssistant, entry: EssentConfigEntry) -> bool:
+    """Set up Essent from a config entry."""
+    coordinator = EssentDataUpdateCoordinator(hass, entry)
+    await coordinator.async_config_entry_first_refresh()
+
+    # Start independent schedules for API fetch and listener updates
+    # These will continue running regardless of API success/failure
+    coordinator.start_schedules()
+
+    entry.async_on_unload(coordinator.async_shutdown)
```

Comment:
Why do we have this?

Reply:
The coordinator uses custom scheduling instead of update_interval for two reasons:

1. API fetch: Runs hourly at a random minute offset (0-59) to spread load across all users
2. Listener tick: Runs exactly on the hour to advance sensors to the next tariff using cached data (no API call)

These timers are registered via async_track_point_in_utc_time and need to be cancelled on unload.

Is there a better pattern for this, or a way the base coordinator can handle this?

Reply:
If update interval is set to an hour it will just update from where it started, it wouldn't bunch people together. 

But it would cause unnecessary API calls as we already have enough data for the same day